objects/src/accounts/component/template/storage_entry/mod.rs

```diff
@@ -0,0 +1,715 @@
+use alloc::{boxed::Box, string::String, vec::Vec};
+
+use vm_core::{
+    utils::{ByteReader, ByteWriter, Deserializable, Serializable},
+    Word,
+};
+use vm_processor::{DeserializationError, Digest};
+
+mod word;
+pub use word::*;
+
+use super::AccountComponentTemplateError;
+use crate::accounts::{StorageMap, StorageSlot};
+
+mod template_key;
+pub use template_key::{TemplateKey, TemplateValue};
+
+mod init_storage_data;
+pub use init_storage_data::InitStorageData;
+
+// STORAGE ENTRY
+// ================================================================================================
+
+/// Represents a single entry in the component’s storage layout.
+///
+/// Each entry can describe:
+/// - A value slot (single word or multiple words).
+/// - A map slot (key-value map that occupies one storage slot).
+/// - A multi-slot entry (spanning multiple contiguous slots, with multiple values).
```

Comment:
Would it make sense to mention that multi-slot entries are only for values (e.g., they can't contain maps).

Reply:
Added this